### ISSUE TITLE
Extension Doesn't seem to be working in Godot 3.2.3RC4

### ISSUE THREAD

**Issue by Alriightyman** [godotengine/godot-csharp-visualstudio#10]
**OS/device including version:**
Windows 10

**Issue description:**
1. Installed Godot CSharp extension
2. Create new project in Godot 3.2.3 RC4
3. Add a Node into the Scene
4. Attach new C# script.
5. Visual Studio opens

Expected: 
Expected to see `Play In Editor`, `Launch`, or `Attach`

Actual:
Only see the Project name and it will not launch instead throwing an error of `A project with an Output Type of Class Library cannot be started directly`

**Screenshots of issue:**
![image](https://user-images.githubusercontent.com/6556309/91474883-d66e0a00-e868-11ea-91f5-1b1b79abcb1e.png)
![image](https://user-images.githubusercontent.com/6556309/91474990-f00f5180-e868-11ea-84bc-02eb257042c6.png)
![image](https://user-images.githubusercontent.com/6556309/91475035-00273100-e869-11ea-9d13-1951ebbbe41c.png)

**NOTES**
This worked just fine when using RC3.  My guess is that RC4 upgraded to the csproj sdk format and that broke compatibility here.

**Comment by neikeq** [comment]
I'll have a look. The ProjectTypeGuid was moved to the Sdk. I hope Visual Studio can support that.

**Comment by akien-mga** [comment]
I had the same issue yesterday with a pre-RC 5 test build, and I expect the newly released 3.2.3 RC 5 still has the same issue.

**Comment by MrZak-dev** [comment]
I confirm this issue  , Windows 3.2.3 RC 5

**Comment by toddgibson** [comment]
I was really excited about this and installed 3.2.3 RC 6 and am experiencing the same issue.

**Comment by 31** [comment]
I dug into this, and I got *something* working again, but with a significantly degraded experience. 😕 Basically, I made a button that assumes the current startup project is a Godot project and launches the debugging session.

---

The problem seems to be the new project system, and not using `ProjectTypeGuid` at all anymore. The first hint I saw towards this was https://github.com/dotnet/project-system/issues/232:

> #### ~ProjectTypeGuids property~
> ```
> <ProjectTypeGuids>{3AC096D0-A1C2-E12C-1390-A8335801FDAB};{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}</ProjectTypeGuids>
> ```
> 
> Used to indicate the project factories that should be used to open a project inside Visual Studio.
> 
> _CPS completely removes the need for this, and this can be safely removed for projects that have opt'd into CPS._

There's also this chart indicating a very different new system in use for .NET-Core-style projects: https://github.com/microsoft/VSProjectSystem#compare-mpfproj-and-visual-studio-project-system

> Criteria | MPFProj | Visual Studio Project System (this repo)
> -- | -- | --
> Maturity | Release | Preview
> Breaking changes in major updates | None | Expected
> [...] | [...] | [...]
> Used by project systems | Desktop C#/VB/F# | .NET Core C#/VB/F# and C++
> [...] | [...] | [...]

They have a doc page for project flavoring: https://github.com/microsoft/VSProjectSystem/blob/master/doc/overview/extend_an_existing_project_type.md. But... as of writing, it's TODO 😢:

> Extend an existing project type
> ===============================
> 
> TODO

I wanted to try out using the new system, but I hit a wall trying to find docs or any kind of example that resembles what the Godot extension needs. Maybe someone will have more success than me here.

---

On to what I did... I added a basic `Command` template and wired in a few methods to assume the current default project is a Godot project, register it, and launch the debugging session:

> ![image](https://user-images.githubusercontent.com/331300/93715820-48b4d000-fb31-11ea-9370-30a37b736574.png)

Branch/diff: [button-assume-godot](https://github.com/godotengine/godot-csharp-visualstudio/compare/master...31:button-assume-godot)

This is my first time doing anything beyond F5ing a VS extension project, so I don't have the experience to polish this up into something that might be usable broadly. Maybe the general approach of simply having our own set of launch buttons might be good enough for now? It's good enough for me, and still *way* better than trying to use the Mono debugger manually.

**Comment by mudbug-z** [comment]
I noticed that in the csproj file the Godot SDK is specified: "<Project Sdk="Godot.NET.Sdk/3.2.3">"
Could that be used to determine if the project is a Godot project?

**Comment by wihrl** [comment]
But will it ever recover?

**Comment by neikeq** [comment]
Sorry for the delayed reply. It's as @31 said, Visual Studio uses a new project system for Sdk style projects.
I've tried about everything to get the extension working with the new VS project system, but nothing seems to work. In fact not even the [samples](https://github.com/microsoft/VSProjectSystem/tree/master/samples) worked (throwing composition exceptions that seem to indicate some of the `DebugLaunchProviderBase` imports cannot be satisfied).
At this point I'm not sure if right now this new project system support the extensibility we need.

As that problem is unlikely to be solved soon, we will need to go with a workaround. I'll make the extension load always to have Godot special code completion work, and I will add commands for launching the debugger similarly to @31's workaround.

That should be enough to make the extension usable again, even if it's a degraded experience. At least until we can make this work with the new project system.

**Comment by neikeq** [comment]
By the way, there's a workaround you can use today to make the extension work. The idea is to use an old style C# project as placeholder to trigger the extension.

Add a new C# project to your solution. Make sure it's an old style project, not Sdk style. Add the `ProjectTypeGuids` containing the Godot GUID, and add a project reference to your actual game project. Here's an example:

```csproj
<?xml version="1.0" encoding="utf-8"?>
<Project ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
  <Import Project="$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props" Condition="Exists('$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props')" />
  <PropertyGroup>
    <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
    <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
    <ProjectGuid>{F97F64DA-67E2-448D-8516-2F7427C8B882}</ProjectGuid>
    <OutputType>Library</OutputType>
    <RootNamespace>StubProj</RootNamespace>
    <AssemblyName>StubProj</AssemblyName>
    <TargetFrameworkVersion>v4.7.2</TargetFrameworkVersion>
    <FileAlignment>512</FileAlignment>
    <AutoGenerateBindingRedirects>true</AutoGenerateBindingRedirects>
    <Deterministic>true</Deterministic>
    <ProjectTypeGuids>{8F3E2DF0-C35C-4265-82FC-BEA011F4A7ED};{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}</ProjectTypeGuids>
  </PropertyGroup>
  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' ">
    <PlatformTarget>AnyCPU</PlatformTarget>
    <DebugSymbols>true</DebugSymbols>
    <DebugType>full</DebugType>
    <Optimize>false</Optimize>
    <OutputPath>bin\Debug\</OutputPath>
    <DefineConstants>DEBUG;TRACE</DefineConstants>
    <ErrorReport>prompt</ErrorReport>
    <WarningLevel>4</WarningLevel>
  </PropertyGroup>
  <ItemGroup>
    <ProjectReference Include="..\MyGame.csproj">
      <Project>{5957a55d-a731-4684-85cb-0bf023e5d393}</Project>
      <Name>MyGame</Name>
      <Private>False</Private>
    </ProjectReference>
  </ItemGroup>
  <Import Project="$(MSBuildToolsPath)\Microsoft.CSharp.targets" />
</Project>
```

Change the path, guid and name of the `ProjectReference` to point to your game project (or just remove that part and add the project reference from Visual Studio).

Set this project as the "Startup Project" in Visual Studio, either from the toolbar next to the launch button or right click the project and "Set as Startup Project".

That should be enough to make it work. The launch button should be "Play in Editor" again.

The reason the placeholder project needs to reference your game project is because otherwise VS shows "No compatible code running" when a breakpoint hits.

**Comment by juanpaexpedite** [comment]
![imagen](https://user-images.githubusercontent.com/5947026/98641841-506c4600-2324-11eb-84db-b2573bd22971.png)

I have done the @neikeq workaround and now I have that issue. What option do I have to do in Godot to make it work?

**Comment by neikeq** [comment]
@juanpaexpedite You should have a Godot editor instance open editing the same project. If you already have and still get that error, then it's a bug. But that would be a different issue to this one.

**Comment by spacechase0** [comment]
I noticed that editing when paused & continuing doesn't appear to actually take the changes (athough breakpoints appear to work fine). (I can also edit the files when the game isn't paused, like other things that work with edit & continue.) Is this a limitation of the addon, or part of this issue?

**Comment by AnderssonPeter** [comment]
Skimmed thru the comments but are sadly did find any thing that could work.
I installed the latest version of the addon (in the release tab) 
1. created a new project in godot 3.2.3
2. added a root scene 
3. attached a .cs file to it
4. Opened the cs file from godot (This opens visual studio)
But no options show up
![image](https://user-images.githubusercontent.com/1308309/103568571-10f64c00-4ec6-11eb-9fa7-4cbe9127a2cb.png)

```xml
<Project Sdk="Godot.NET.Sdk/3.2.3">
  <PropertyGroup>
    <TargetFramework>net472</TargetFramework>
  </PropertyGroup>
</Project>
```
Am i forced to use the old ,csproj format to make this work?

**Comment by piratesephiroth** [comment]
> 
> Am i forced to use the old ,csproj format to make this work?

yeah, at the moment you have to use the workaround posted above by @neikeq

**Comment by BaronKiko** [comment]
I can confirm the workaround still works. I feel like a better solution. and something that's needed regardless. is a generate vs project button in godot. Then it can just generate the proxy project for you.

**Comment by 31** [comment]
I've pointed a few people at this issue who haven't been able to understand how to apply the workaround. Maybe new to .NET, or new to VS. So, I decided to try it out and document a full process to get the workaround in place.

Step by step guide: https://gist.github.com/31/302bab2630ddf8d90ab4aec0d358b538

**Comment by magian1127** [comment]
vs 2019 16.9.x 
if use the above method, intelligent prompt inoperative

**Comment by piratesephiroth** [comment]
>  intelligent prompt

what's intelligent prompt?
Intelisense is working just fine in VS 16.9.2

![image](https://user-images.githubusercontent.com/10672233/113030738-4aa4bf00-9164-11eb-9640-d75d45ff1481.png)

**Comment by pjmavcom** [comment]
Trying the @31 workaround guide got me to this error:

Method not found: 'Void Mono.Debugging.Evaluation.ObjectValueAdaptor.set_DebuggerSession(Mono.Debugging.Client.DebuggerSession)'.

Any idea what I did wrong? Thanks!

**Comment by piratesephiroth** [comment]
Here's my tutorial:
Get the snippet neikeq posted above and paste it into a new text file.

Now you have edit this "ProjectReference" block:
```xml
    <ProjectReference Include="..\MyGame.csproj">
      <Project>{5957a55d-a731-4684-85cb-0bf023e5d393}</Project>
      <Name>MyGame</Name>
      <Private>False</Private>
    </ProjectReference>
```  

Change the Include attribute, from "..\MyGame.csproj" to a relative path that points to your game's project.
Change the Project to your project's  ProjectGuid.
Change the Name element from "MyGame" to "Stub" or whatever.
Save the file as Stub.csproj or whatever.

Now open your project's solution file in Visual Studio.
Right click the solution in Solution Explorer and select "Add" - > "Existing Project", then select your new Stub project file.

Now right-click the Stub project in Solution Explorer and select "Set as Startup project".
(you can also do that from the solution's properties)

**Comment by pjmavcom** [comment]
@piratesephiroth Thanks for the quick reply! Im still getting the same error. Wondering if I just messed it all up trying a few different things before coming to this? I followed this first. 
https://docs.godotengine.org/en/stable/getting_started/scripting/c_sharp/c_sharp_basics.html#configuring-vs-2019-for-debugging
The GUID for the project is in the .sln file, Project section, the second set of numbers on the line with my project name right?
If that shouldn't mess it up, then is there something in Godot I'm not doing? I've got VS selected as the external editor. Haven't messed with the project settings.

**Comment by piratesephiroth** [comment]
You should get the GUID from your project file directly to avoid any mistakes.

**Comment by mudbug-z** [comment]
In case anyone is interested, JetBrains Rider works GREAT with Godot mono out of the box.  Of course, it isn't free.  For me it is worth the cost.

**Comment by magian1127** [comment]
> ![image](https://user-images.githubusercontent.com/10672233/113030738-4aa4bf00-9164-11eb-9640-d75d45ff1481.png)

Excuse me for my poor English
Intellisense not working 
Dereference Stub.csproj Intellisense is working 

 my VS 16.9.2 not  working 
 VS 16.7.x working

**Comment by pjmavcom** [comment]
You know what? I forgot to mention the main project .csproj file doesn't have a GUID in it:
<Project Sdk="Godot.NET.Sdk/3.2.3">
  <PropertyGroup>
    <TargetFramework>net472</TargetFramework>
    <RootNamespace>Platformer</RootNamespace>
  </PropertyGroup>
</Project>
The other projects I've made have them. I'm going to play around with it and see if I can figure out why, and see if that is the problem maybe.
As for Rider @mudbug-z, I have it and Visual Studio/Resharper both. I like VS just a bit more, some hotkey or something Rider doesn't have, I can't remember. But it's getting to the point where I will take just works over whatever I used to like more lol

Alright, last ditch effort, then I'll start asking about how to get Rider to work. LOL
I just started a brand new project in Godot. Empty. Made an object, attached a C# script. It created an .sln and .csproj, but the project file is as the last one. No GUID and just those 6 lines. Well, 5, doesn't have a namespace yet. I tried 3.3 rc7 mono in case mine wasn't new enough. Same outcome. Also tried setting this up with another project that had a full project file. Same error.

**Comment by 31** [comment]
> You know what? I forgot to mention the main project .csproj file doesn't have a GUID in it:

This is normal as of Godot 3.2.3+, since it uses the new project system. I didn't need any GUID in my `ProjectReference` for the workaround to work for me.

**Comment by pjmavcom** [comment]
@31 Yeah I remember seeing that somewhere now. I'm going to get Rider setup, but I'll still try and figure out the VS thing if anyone's got any ideas about it.

**Comment by piratesephiroth** [comment]
OK, I just created a new c# project in Godot 3.3.rc7 and VS 2019 16.9.3.
It worked just fine.
The "Play in Editor" button became the default action as soon as I set the Stub project to Startup Project and everything functioned properly. 
I updated the tutorial with info regarding the ProjectGUID.

I don't understand what's going wrong for your guys.
Are you sure you added a proper path to your actual project in the stub one?

**Comment by piratesephiroth** [comment]
Yeah, it worked here both next to the main project and in a subfolder.

**Comment by pjmavcom** [comment]
Well, being Windows and Visual Studio, I was probably holding the mouse upside down when I installed it. Not a big deal for me since Rider works now. When I get around to doing fresh installs I'll try again, but it's probably some random config I didn't know I setup or something.

**Comment by piratesephiroth** [comment]
> 
> 
> Well, being Windows and Visual Studio, I was probably holding the mouse upside down when I installed it. Not a big deal for me since Rider works now. When I get around to doing fresh installs I'll try again, but it's probably some random config I didn't know I setup or something.
> EDIT: Well Visual Studio updated finally and I get a different error:
> ![Clipboard03](https://user-images.githubusercontent.com/10098645/113194479-cebf7b00-9226-11eb-9893-941a672313b9.jpg)
> I have installed the VSMonoDebugger extension. Anything else supposed to be installed?

Maybe it's that extension that is causing trouble then, because I don't have it installed.
The only mono-related extension I have installed ia a fork of MonoHelper that I use to build some small unrelated projects.

**Comment by pjmavcom** [comment]
Alright! That was it. This link: https://docs.godotengine.org/en/stable/getting_started/scripting/c_sharp/c_sharp_basics.html#configuring-vs-2019-for-debugging does not work with this fix. Uninstalled VSMonoDebugger and it started right up.
Thank you @piratesephiroth and @31!

**Comment by piratesephiroth** [comment]
> 
> 
> Alright! That was it. This link: https://docs.godotengine.org/en/stable/getting_started/scripting/c_sharp/c_sharp_basics.html#configuring-vs-2019-for-debugging does not work with this fix. Uninstalled VSMonoDebugger and it started right up.
> Thank you @piratesephiroth and @31!

Cool! I added that to my tutorial.

**Comment by SimonasLetukas** [comment]
Hello, not sure if this is connected to this extension, but after doing what @piratesephiroth suggested I get this error upon pressing "play in editor":
![image](https://user-images.githubusercontent.com/4127189/113762286-1c554f80-9721-11eb-9217-a2e52cd4ae04.png)

**Comment by piratesephiroth** [comment]
> 
> 
> Hello, not sure if this is connected to this extension, but after doing what @piratesephiroth suggested I get this error upon pressing "play in editor":

Yeah, that package is required by this extension. I'm not sure why it's missing though.
The Godot project itself doesn't require that.

**Comment by SimonasLetukas** [comment]
What could be done in such a case then? I tried adding the package Clide.Windows 4.0.17 (couldn't find 4.0.16 in NuGet) both in the main project and the stub project but still cannot launch the debugging. :/

**Comment by Barste** [comment]
Hello,

I successfully managed to apply the workaround by following the steps described by @piratesephiroth, and using the play in editor to launch Godot works perfectly fine, but my issue is that as soon as I add the stub in the solution, the autocompletion stops functioning.

Is there anything specific to do in order to avoid breaking the completion?

I am using Godot 3.2.3 and visual studio 2019 version 16.9.4
Any help would be greatly appreciated!

**Comment by Xyotic** [comment]
I managed to get this extension to work with @31 workaround.
But I had to add a project reference to my main projects solution.
Otherwise the brakepoints didnt hit.

**Comment by azbay** [comment]
Exciting news, _maybe_.

To update the research that @31 did (over a year ago now, lol) there have been some potentially useful things I've found. I'm not experienced enough to figure out the details of implementing this in a way that won't drive me insane, but hopefully someone else is capable.

The most useful resource (probably, at least in the meantime) is [this section of the CPS repo](https://github.com/microsoft/VSProjectSystem/tree/cb8c730eef9d4f750cb219ff24f0bf66be429e2c/samples/CpsExtension) which seems to detail exactly what we need. Of course, the sample itself was last updated **_3 years ago_**, so that might throw a wrench into the works. 

The only other caveat is that they (I think) use a NuGet package as the qualifying parameter that makes the debug option show up, so I'm not certain if you could switch that to target the new SDK parameter in .csproj (or however else the Godot SDK project type manifests itself).

The full documentation for CPS is found [here](https://github.com/microsoft/VSProjectSystem/blob/master/doc/Index.md), which should help with any other issues that crop up, especially regarding the outdated-ness of the sample plugin.

**Comment by neikeq** [comment]
I remember trying the CpsExtension sample without success (it either didn't compile or MEF threw exceptions due to missing stuff).
I might give it another try soon with the information here (it seems they added this page a few months ago): https://github.com/dotnet/project-system/blob/main/docs/launch-profiles.md

**Comment by Xyotic** [comment]
@neikeq could you share your progress/the things you already tried? 
Maybe on a seperate branch, so the community could take a look on that issue as well and wont have to start from scratch.

**Comment by GeorgeS2019** [comment]
@Xyotic I am totally new to this development. 
However, I routinely use GodotXunit for test driven, I am sharing the link here to integrate Visua studio 2022 [here](https://github.com/fledware/GodotXUnit/issues/9#issuecomment-980537087).

Contribution is welcome!

**Comment by xchellx** [comment]
This does not work anymore (unless you want to be restricted to .NET Standard 2.0) `Project '..\CMDL_View.csproj' targets 'net6.0'. It cannot be referenced by a project that targets '.NETFramework,Version=v4.7.2'.`

**Comment by xchellx** [comment]
This does not work anymore (unless you want to be restricted to .NET Standard 2.0) `Project '..\MyGame.csproj' targets 'net6.0'. It cannot be referenced by a project that targets '.NETFramework,Version=v4.7.2'.`

**Comment by userqt** [comment]
Hello there, its 2022 and I am not able to see the Play in Editor button. Has this been fixed or we have to use some workarounds still? If yes, which one?

I am using visual studio 2022 and Godot 3.4.4 (Mono version (C# support)). Any help would be appreciated.

**Comment by PandoCyclone** [comment]
it is Godot 3.5 now and no fix for this :(

**Comment by Alriightyman** [comment]
@neikeq Just a question related to this bug, since .net6 was merged into Godot 4, will Godot 4 suffer from this bug as well or was work done to solve this issue without the above workaround?

**Comment by Alriightyman** [comment]
Ah, I see. I didn’t get that far in the conversation. So, in that case, the extension isn’t used then. Thanks for the info.

**Comment by Xyotic** [comment]
Found something new. Maybe we this could help to solve the sdk style problem...
https://docs.microsoft.com/en-us/visualstudio/ide/customize-build-and-debug-tasks-in-visual-studio?view=vs-2022

Maybe we could use a launch.vs.json to set the different debug targets.
And then we would need to communicate it back to the extension.

Mabe there are events for that?

**Comment by GeorgeS2019** [comment]
Look promising

**Comment by GeorgeS2019** [comment]
This [PR ](https://github.com/godotengine/godot-csharp-visualstudio/pull/32)implements a way to define the Godot.exe path and execute it in a reliable way than the previous StubProj. All it needs is to deliver the parameters for different scenarios of debugging.  neikeq has different messaging system to capture the standard and errors outputs for display in VS2022

**Comment by Xyotic** [comment]
The code to define the .exe path could be used to create the .csproj entry.

But other than that I think these two approches are too different to be useful for eachother.

I didn't try if my solution works with breakpoints tho. Will do that later.
But according to this, it should work?

> Currently, the way to debug C# in Godot 4 with Visual Studio is as I explained [here](https://github.com/godotengine/godot/pull/64089#issuecomment-1209131428):

**Comment by Xyotic** [comment]
Update:
Using "StartAction" + "StartProgram" in .csproj in combination with launchSettings.json do hit breakpoints in 4.0.

The next steps would be to check if the extension can interrupt startup and get information from the launchSettings. Maybe via environment variables... This way we could implement an "attach" launch profile.

**Comment by Xyotic** [comment]
I did some more digging and tried to make the [Csp sample](https://github.com/microsoft/VSProjectSystem/tree/cb8c730eef9d4f750cb219ff24f0bf66be429e2c/samples/CpsExtension) work @azbay mentioned.

Like @neikeq, it did build out of the box, but I managed to make it build.
Basically all you have to do is to update the nuget packages inside this sample project.
If you go with the preview packages (which are pretty current and just got some new updates), you'll also have to change the target framework of the sample to 4.7.2

Now It should build. The extension seems to load (it hits the breakpoint inside the CustomDebugger.cs constructor), but it doesnt add the new "CustomDebugger" Debug option to the Godot project... 
I've also followed the instructions and did add the nuget to the godot project.

I think it has something todo with the nuget itself. Inside the "CpsExtension.Nuget.targets" the CpsExtensionDesignTimeTargetsPath uses a condition which cant be fullfilled. I tried to remove the condition but it also didnt show up the CustomDebugger...

Maybe someone else got more ideas?

**Comment by GeorgeS2019** [comment]
> But other than that I think these two approaches are too different to be useful for each other.

Can you clarify are you relying on StubProj? 
I could not figure out how the StubProj works by launching Godot. 

However, the new CommandLine to launching Godot seems clear to follow.

What needs to be done is to connect the Godot VS2022 extension Messaging to receive the standard and error outputs of running an external command (e.g. Godot.exe). These are the information that help to communicate with VS2022 where are the error and what are the error messages.

**Comment by GeorgeS2019** [comment]
Is this well documented? There is an option which is not StubProj and the latest commandline?

Can u share the links I could follow this Third method?

**Comment by Xyotic** [comment]
Follow my findings I mentioned above.
Basically you only need to add this:

```xml
 <PropertyGroup>
    <StartAction>Program</StartAction>
    <StartProgram>C:\Path\To\Godot.exe</StartProgram>
    <StartArguments>--path C:\Path\To\Godot\Project\</StartArguments>
  </PropertyGroup>
```

to your .csproject and you can start debugging.
If you want to add the differen launch profiles you can setup the launchsettings.json.

But currently I'm more interested in getting the csp sample to work

**Comment by Logopher** [comment]
Improved with MSBuild properties (I defined GODOT_EXE as an environment variable):

```
  <PropertyGroup>
    <StartAction>Program</StartAction>
    <StartProgram>$(GODOT_EXE)</StartProgram>
    <StartArguments>--path "$(MSBuildProjectDirectory)"</StartArguments>
  </PropertyGroup>
```

**Comment by GeorgeS2019** [comment]
This should be closed...Godot 3.
Ask in disvord
